src/Utils/errors.tsx

```diff
@@ -4,22 +4,26 @@ export class NetworkError extends Error {
   response: any
 }
 
-interface ErrorInfo {
-  componentStack: string
+export class ErrorWithMetadata extends Error {
+  metadata: object
+  decoratedMessage: string
+  constructor(message, metadata = {}) {
+    super(message)
```

Comment:
@alloy I decided not to make the `decoratedMessage` the actual error message since the metadata can be very long and obscure what's going on.